ath79:enable link state reporting for eth1 on Qxwlan e750a/e600g/e600gac

zhangsd20 opened the pull request:
Enable link state reporting or the Fast Ethernet port attached through
the built-in switch, so it can generate netifd and hotplug events as well.

I tested it on several devices with different CPUs and they all worked really
well after adding the code to solve the link state reporting issue.

Test Device:
Qxwlan e750a (soc: ar9344)
Qxwlan e600g/gac (soc :qca9531)
Qxwlan ml181 (soc: qca9561)
ML181 currently exists only in local sdk and will be committed to OpenWRT

My code is based on: https://github.com/openwrt/openwrt/pull/9971

Signed-off-by: 张 鹏 [sd20@qxwlan.com](mailto:sd20@qxwlan.com)

Leo-PL committed:
ath79: ag71xx: use "builtin-switch" property to set link configuration

To support reporting link state of PHYs attached to built-in switch,
add a device tree knob which allows to force 1000Mbps/FD mode,
which is the link mode between eth1 MAC and the on-chip switch, even if
no "fixed-link" node is present. Re-use the "builtin-switch" name
already used in respective MDIO nodes.

This way, a phy-handle property can be added to eth1 node, and devices,
which have a single port attached through the built-in switch,
can report proper link state of that to userspace.

To perform that, one needs to delete the 'fixed-link' node and map the
correct swphy node to 'phy-handle' property. One of those is still
required to be present in the eth1 node.

Signed-off-by: Lech Perczak <lech.perczak@gmail.com>

Leo-PL committed:
ath79: dts: set builtin-switch for SoCs with fixed-link on eth1

Introduce the property from previous commit on the SoCs which use a
fixed 1000FD link to their internal switch. This way, devices which have
a single port attached through it can drop the "fixed-link" node if
needed, and attach proper phy-handle provided by built-in switch to
the port, to report link status information on userspace.

AR7100 is skipped intentionally, because its connection to built-in
switch isn't a fixed-link.

Signed-off-by: Lech Perczak <lech.perczak@gmail.com>

Leo-PL committed:
ath79: enable link state reporting for eth1 on Ruckus ZF7372

Now, that the support for ZF7372 is merged, enable link state reporting
for the Fast Ethernet port attached through the built-in switch, so
it can generate netifd and hotplug events as well, for example -
- to control DHCP client.

Signed-off-by: Lech Perczak <lech.perczak@gmail.com>

Leo-PL commented:
NAK. This copies #9971 verbatim, removing all of documentation in commit messages, squashing everything together and removing my copyright. My examples for Ruckus radios are also removed.

zhangsd20 commented:
Sorry I don't know how I should modify it. Please tell me how it should be done

zhangsd20 commented:
Thanks!
Sorry I don't know how I should modify it. 
Please tell me how it should be done.

Leo-PL commented:
I think the easiest is to  add my repo as remote to your Git repo, then do the rebase and cherry-pick your changes to device tree files.
```
git remote add Leo-PL https://github.com/Leo-PL/openwrt.git
git fetch Leo-PL
git checkout ath79-builtin-switch-link-state
git rebase main
git cherry-pick <commit-ish> # you changes to device trees here
```
And this should do the trick. Alternatively, you can download patches from Github directly by navigating to the commit, and adding a `.patch` suffix, then they can be applied with all the details just by `git am <patch_file>`

zhangsd20 commented:
git cherry-pick xxx
I'm having trouble executing this instruction.
Do I need to re-add my device tree commit on top of that?
  


在2023年06月12 15时09分，"Lech ***@***.***>写道：


I think the easiest is to  add my repo as remote to your Git repo, then do the rebase and cherry-pick your changes to device tree files.
git remote add Leo-PL https://github.com/Leo-PL/openwrt.git
git fetch Leo-PL
git checkout ath79-builtin-switch-link-state
git rebase main
git cherry-pick <commit-ish> # you changes to device trees here
And this should do the trick. Alternatively, you can download patches from Github directly by navigating to the commit, and adding a .patch suffix, then they can be applied with all the details just by git am <patch_file>
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Leo-PL commented:
I think you need to separate changes to actual Qxwlan device trees into separate commit first, or apply them manually - it looks, like you have squashed changes for device trees, includes (the generic part) and for ag71xx driver in the same commit.)

zhangsd20 commented:
I've gone back to the original version and only made the device tree commit













在2023年06月12 16时46分，"Lech ***@***.***>写道：


I think you need to separate changes to actual Qxwlan device trees into separate commit first, or apply them manually - it looks, like you have squashed changes for device trees, includes (the generic part) and for ag71xx driver in the same commit.)
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zhangsd20 committed:
ath9k: enable link state reporting for eth1 on Qxwlan e750a/e600g/e600gac

Now, that the support for e750a/e600g/e600gac is merged, enable link state
reporting for the Fast Ethernet port attached through the built-in switch,
so it can generate netifd and hotplug events as well, for example -
- to control DHCP client.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>

zhangsd20 commented:
I refixed the PR, is this possible?https://github.com/openwrt/openwrt/pull/9971/commits/74c976f5b21fec54bafc436993f692d8530ca099













在2023年06月12 16时46分，"Lech ***@***.***>写道：


I think you need to separate changes to actual Qxwlan device trees into separate commit first, or apply them manually - it looks, like you have squashed changes for device trees, includes (the generic part) and for ag71xx driver in the same commit.)
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Leo-PL commented:
Hi,
No. You did reset the commit authorship in the process, again. Please do not manually apply patches, use solely `git am` for that.

zhangsd20 commented:
I mean I reworked my PR and separated the device tree.
And "git cherry-pick <commit-ish> " is ok.
Then how should I do it, push it directly?

Leo-PL commented:
Yeah, you can use `git push --force` to replace contents in this PR.

zhangsd20 commented:
Sorry,I can't complete the push.
remote: Support for password authentication was removed on August 13, 2021.
fatal: Authentication failed for 'https://github.com/Leo-PL/openwrt.git/'

Leo-PL commented:
Of course you cannot push to my branch. You need to reset this branch to the new contents, and then push it here.

zhangsd20 commented:
How should I operate?
please help me!

Thanks!

Leo-PL commented:
```
git  checkout -b main_old
git checkout main
git reset --hard ath79-builtin-switch-link-state
git push --force-with-lease
```
This will keep the original branch at main_old and push new changes here.

Leo-PL commented:
All right, now it makes sense. Thanks!

zhangsd20 commented:
Ok, Thank you very much!

Neustradamus commented:
@zhangsd20: Any progress on it?

Replaces:
- https://github.com/openwrt/openwrt/pull/12830

Leo-PL commented:
@Neustradamus please check my previous PR, where I added more device. I'll cherry-pick required changes for Qxwlan stuff, too - thanks for noticing.
